### ISSUE TITLE
`normalize` on `SphericalHarmonics` is broken

### ISSUE THREAD

**Issue by rwkeane** [NVIDIA/cuEquivariance#201]
I was testing, and I found that spherical harmonics don't behave correctly for the non-normalized case (at least on the CPU). Library version is `0.7.0`

Repro case is as follows (with `pytest`):

```
def test_minimal():
    normalized_module = SphericalHarmonics([0], normalize=True, device="cpu")
    non_normalized_module = SphericalHarmonics([0], normalize=False, device="cpu")
    value = torch.randn(1, 3)

    assert normalized_module.forward(value).isclose(torch.tensor([1], dtype=torch.float32))  # This works
    assert non_normalized_module.forward(value).isclose(torch.tensor([1.0/math.sqrt(4*math.pi)])) # This doesnt, the module instead returns 1
```

Additionally, the non-normalized case doesn't expose a boolean for the Condon-Shortley convention (at least in my case its needed, but not sure about other people), but this is more of a feature request than a bug
